test/parallel/test-runner-reporters.js

```diff
@@ -92,4 +92,24 @@ describe('node:test reporters', { concurrency: true }, () => {
       assert.strictEqual(child.stdout.toString(), `${filename} {"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":7}`);
     });
   });
+
+  it('should support a custom reporter from node_modules', async () => {
+    const child = spawnSync(process.execPath,
+                            ['--test', '--test-reporter', 'reporter-cjs', 'reporters.js'],
+                            { cwd: fixtures.path('test-runner') });
+    assert.strictEqual(child.stderr.toString(), '');
+    assert.strictEqual(child.stdout.toString(),
+                       'package: reporter-cjs' +
+                       '{"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":7}');
```

Comment:
I think this is re-introducing the flakiness that https://github.com/nodejs/node/pull/45930 has addressed.
```suggestion
    assert.match(
      child.stdout.toString(),
      /package: reporter-cjs {"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":\d+}/
    );
```